### ISSUE TITLE
Feat: Support chrono Date<_> Scalar

### ISSUE THREAD

**Issue by ecklf** [async-graphql/async-graphql#653]
Would be nice to have this scalar out of the box when enabling the `chrono` flag. Useful for APIs that handle Date ranges.

Can try to PR if needed.

**Comment by ecklf** [comment]
> I'm not sure about this, we'll often want to adjust the date formatting depending on the domain & targets. If we give an implementation for a Date, we will have to choose a way to format this date that will constrain all users of the library.
> 
> **Edit**: Well, there is already some implementation of this kind for datetime and naivedate, with the rfc3339 and another format but I keep thinking it shouldn't be done for date 🤷‍♂️

Yeah only suggesting this because it has already been done for `DateTime`. I am not sure if GraphQL has a standard of how date formats should look like. If there is it would be a viable thing to do imo.

**Comment by sunli829** [comment]
> @sunli829 cool, will look into it. What do you think about adding the ability to switch between different `Date` scalar modes? I know this from https://typegraphql.com/docs/scalars.html#date-scalars. Just not sure how one would achieve this the best way in Rust.

It may be necessary to add some feature gates.
